frame/vesting/src/benchmarking.rs

```diff
@@ -161,65 +188,184 @@ benchmarks! {
+		assert_eq!(System::<T>::block_number(), T::BlockNumber::zero());
+		assert_eq!(
+			Vesting::<T>::vesting_balance(&caller),
+			Some(expected_balance),
+			"Vesting balance should equal sum locked of all schedules",
+		);
+		assert_eq!(
+			Vesting::<T>::vesting(&caller).unwrap().len(),
+			s as usize,
+			"There should be exactly max vesting schedules"
+		);
+	}: merge_schedules(RawOrigin::Signed(caller.clone()), 0, s - 1)
+	verify {
+		let expected_schedule = VestingInfo::new::<T>(
+			T::MinVestedTransfer::get() * 2u32.into(),
+			per_block_duration_20 * 2u32.into(),
+			1u32.into(),
+		);
+		let expected_index = (s - 2) as usize;
+		assert_eq!(
+			Vesting::<T>::vesting(&caller).unwrap()[expected_index],
+			expected_schedule
+		);
+		assert_eq!(
+			Vesting::<T>::vesting_balance(&caller),
+			Some(expected_balance),
+			"Vesting balance should equal total locked of all schedules",
+		);
+		assert_eq!(
+			Vesting::<T>::vesting(&caller).unwrap().len(),
+			(s - 1) as usize,
+			"Schedule count should reduce by 1"
+		);
+	}
+
+	unlocking_merge_schedules {
+		let l in 0 .. MaxLocksOf::<T>::get() - 1;
+		let s in 2 .. T::MaxVestingSchedules::get();
+
+		let caller: T::AccountId = account("caller", 0, SEED);
+		let caller_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(caller.clone());
+		// Give target other locks.
+		add_locks::<T>(&caller, l as u8);
+		// Add max vesting schedules.
+		let mut expected_balance = add_vesting_schedules::<T>(caller_lookup.clone(), s)?;
+		// Values of the schedules added by `add_vesting_schedules`.
+		let transfer_amount = T::MinVestedTransfer::get();
+		let per_block_duration_20 = transfer_amount.checked_div(&20u32.into()).unwrap();
+
+		// Go to half way through all the schedules duration. (They all start at 1, and have a duration of 20).
+		System::<T>::set_block_number(11u32.into());
+		// We expect half the original locked balance.
+		expected_balance = expected_balance / 2u32.into();
+		assert_eq!(
+			Vesting::<T>::vesting_balance(&caller),
+			Some(expected_balance),
+			"Vesting balance should reflect that we are half way through all schedules duration",
+		);
+		assert_eq!(
+			Vesting::<T>::vesting(&caller).unwrap().len(),
+			s as usize,
+			"There should be exactly max vesting schedules"
+		);
+	}: merge_schedules(RawOrigin::Signed(caller.clone()), 0, s - 1)
+	verify {
+		let expected_schedule = VestingInfo::new::<T>(
+			transfer_amount,
+			per_block_duration_20 * 2u32.into(),
+			11u32.into(),
+		);
+		let expected_index = (s - 2) as usize;
+		assert_eq!(
+			Vesting::<T>::vesting(&caller).unwrap()[expected_index],
+			expected_schedule,
+			"New schedule is properly created and placed"
+		);
+		assert_eq!(
+			Vesting::<T>::vesting(&caller).unwrap()[expected_index],
+			expected_schedule
+		);
+		assert_eq!(
+			Vesting::<T>::vesting_balance(&caller),
+			Some(expected_balance),
+			"Vesting balance should equal half total locked of all schedules",
```

Comment:
Why is the expected balance here the same as above?

I see you did expected = expected / 2, but why does that work above? Wouldnt the vesting balance still be full until it is all unlocked?

I think there should be some assert down here that some amount of funds were unlocked to the user?

Reply:
> wouldnt the vesting balance still be full until it is all unlocked?

`merge_schedules` unlocks the vesting balance up through the current block for _all_ schedules. (We do this because we are already iterating through all the schedules and reading/writing locks so it is just as efficient. Plus it makes the logic simpler imo)

So we are able to `expected = expected / 2` because we go to half way through the duration of all schedules and then merge 2, which all unlocks funds for all schedules.

I will add something to clarify for these

Reply:
> I think there should be some assert down here that some amount of funds were unlocked to the user?

Hmm I was trying to use `Balances::usable_balance` to show what got unlocked since `vesting_balance` doesn't reflect the actual locks, but realizing the `LockableCurrency` trait does not have an abstraction equivalent to usable balance so I am not sure how to assert their lock changed without getting messy

Any suggestions?

Reply:
try doing a transfer operation that fails and then passes

Reply:
ce37a9e

Note I just assert `is_err` bc the `assert_noop` requires a concrete error, which we don't have access too for the balances pallet.